geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -335,10 +356,14 @@ protected void runDispatcher() {
       logger.debug("{}: Beginning to process events", this);
     }
 
+    long reAuthenticateWaitTime =
+        getSystemProperty(RE_AUTHENTICATE_WAIT_TIME, DEFAULT_RE_AUTHENTICATE_WAIT_TIME);
+
     ClientMessage clientMessage = null;
+    long wait_for_re_auth_start_time = -1;
```

Comment:
Please consider moving this variable declaration closer to where it's needed. It can moved inside the `while` body: down to just before the `try` block at line 413.

Reply:
It has to be outside of the `while` loop. The same message will be processed by multiple loops if authentication expired and user hasn't re-authenticated back.